src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -1004,4 +1004,18 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
   <Target Name="RunResolvePublishAssemblies" />
 
+  <!--
+    ============================================================
+    _CheckForLanguageAndPublishFeatureCombinationSupport
+    
+    Block unsupported language and feature combination.
+    ============================================================
+    -->
+  <Target Name="_CheckForLanguageAndPublishFeatureCombinationSupport"
+        BeforeTargets="Publish">
```

Comment:
The target that fails first appears to be: "PrepareForPublish"; can we trigger the check before both?
Referring to the test: Microsoft.NET.Build.Tests.GivenThatWeWantToPublishACppCliProject.When_not_referenced_by_csharp_project_it_fails_to_publish

Reply:
However, the Target is missing due to we does not import Microsoft.PackageDependencyResolution.targets for C++ (not used in C++). And it fails to find one of the target there. We already has an empty stub targets file for a similar issue. So I added stub version of ResolveLockFileCopyLocalFiles to it too

Reply:
FYI @nguerrera I added a stub ResolveLockFileCopyLocalFiles to Microsoft.PackageDependencyResolutionStubs.targets since you last review. That target is dependent by Publish.

Reply:
Thanks, @johnbeisner, @wli3 cb5bb53 looks good to me. :)